### PR TITLE
chore: render only env controller action resource if the env stack has no updates 

### PR DESCRIPTION
**Pull request by efekarakus**
Previously, the progress tracker assumed that if the EnvControllerAction is triggered with new Parameters, then the EnvStack gets updated.  
However, that's not necessarily true. The EnvControllerAction can be triggered yet not update the EnvStack resources.

In this change, we're introducing a new component `envControllerComponent` that renders the env stack changes if there are any, otherwise defaults to rendering the env controller action resource.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 1]
```diff
@@ -319,34 +319,42 @@ type envControllerRendererInput struct {
...
-		})
-		return progress.ListeningStackRenderer(streamer, envStackName, in.description, descriptions, in.renderOpts), nil
-	default:
-		logicalID := aws.StringValue(in.change.ResourceChange.LogicalResourceId)
-		return progress.ListeningResourceRenderer(in.serviceStack, logicalID, in.description, progress.ResourceRendererOpts{
-			RenderOpts: in.renderOpts,
-		}), nil
+	workload, err := cf.cfnClient.Describe(in.workloadStackName)
+	if err != nil {
+		return nil, err
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/cloudformation.go:324]: Shall we wrap this error?

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 2]
```diff
@@ -319,34 +319,42 @@ type envControllerRendererInput struct {
...
+	}
+	envStackName := fmt.Sprintf("%s-%s", parseAppNameFromTags(workload.Tags), parseEnvNameFromTags(workload.Tags))
+	body, err := cf.cfnClient.TemplateBody(envStackName)
+	if err != nil {
+		return nil, err
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/cloudformation.go:329]: same for this

**Review by iamhopaul123 — APPROVED** [review]
LGTM just two nits :shipit:

File: internal/pkg/deploy/cloudformation/cloudformation.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go:324]: The cfn package already wraps these errors like this: `fmt.Errorf("describe stack %s: %w", name, err)` so I left it as is. I can do something like `fmt.Errorf("describe stack to render env controller: %w, err)` but it seems unnecessary 🤔

Comment (iamhopaul123) [reply internal/pkg/deploy/cloudformation/cloudformation.go:324]: i see. nvm just ignore this comment

**Review by Lou1415926 — APPROVED** [review]
Looks good to me :shipit:

**Pull request merged**